### PR TITLE
Skip test when git symlinks are not configured

### PR DESCRIPTION
**Pull request by dwoz**
### What does this PR do?

Skips symlink tests when git configuration does not support symlinks.

### What issues does this PR fix or reference?

#46347

### Previous Behavior

Tests would fail if git was cloned without 'core.symlinks=true'

### New Behavior

Tests pass when the config supports them and are skipped with a nice message otherwise.

### Tests written?

No - Fixes existing tests

### Commits signed with GPG?

Yes

### CODE REVIEW

File: tests/unit/modules/test_inspect_collector.py [hunk 1]
```diff
@@ -33,7 +34,23 @@
 from salt.modules.inspectlib.collector import Inspector
 
 
+HAS_SYMLINKS = None
+
+
+def no_symlinks():
```
Comment (dwoz) [inline tests/unit/modules/test_inspect_collector.py]: There may be a more 'salty' way of doing this. I looked around a bit but didn't see one.

Comment (twangboy) [reply tests/unit/modules/test_inspect_collector.py]: There's also an option in the git installer for Windows where you have to enable symlinks on Windows. I don't know how you would check for that.

Comment (dwoz) [reply tests/unit/modules/test_inspect_collector.py]: I'm sure that option will just set the global config option for symlinks. This check will encompass that.

**Review by twangboy — APPROVED** [review]

**Comment by rallytime** [comment]
@dwoz Looks like there are some related test failures here: 

- https://jenkins.saltstack.com/job/PR/job/salt-pr-linode-cent7-py3/3402/
- https://jenkins.saltstack.com/job/PR/job/salt-pr-linode-ubuntu14-n/21069/

**Comment by rallytime** [comment]
@dwoz There's still a test failure on this and a lint error.

**Comment by dwoz** [comment]
re-run py

**Comment by dwoz** [comment]
re-run py

**Comment by rallytime** [comment]
re-run py

**Review by rallytime — APPROVED** [review]

**Pull request merged**